draft-ietf-privacypass-key-consistency.md

```diff
@@ -91,59 +99,97 @@ solutions to this problem.
+two paths as in-band and out-of-band verification. In-band verification is a check
+which is invoked as part of a reliant system. This type of verification is only achieved
+by participants of the reliant system. In contrast, out-of-band verifiability is a check
+that happens outside of a reliant system, i.e., by entities that may not be participants
+of the reliant system. Consistency verification is typically public, meaning that any entity
+with two key sets can verify (global) consistency without requiring knowledge of a
+cryptographic secret.
+
+Reliant systems must also consider agility when trying to achieve consistency. A naive solution to
+ensuring consistent keys is to only use a single, fixed key pair for the entirety of the system.
+Clients can then embed this key into software or elsewhere as needed, without any additional
+mechanics or controls to ensure that other clients have a different key. However, this solution clearly
 is not viable in practice. If the corresponding key is compromised, the system fails. Rotation must
-therefore be supported, and in doing so, users need some mechanism to ensure that newly rotated
-keys are consistent and correct.
+therefore be supported, and in doing so, clients need some mechanism to ensure that newly rotated
+keys are consistent.
+
+Operationally, servers rotating keys may likely need to accommodate distributed system
+state-synchronization issues without sacrificing availability. Some systems and protocols
+may choose to prioritize strong consistency over availability, but this document assumes
+that availability is preferred to total consistency.
 
-Operationally, servers rotating keys may likely need to accommodate
-distributed system state-synchronization issues without sacrificing availability. Some systems and protocols
-may choose to prioritize strong consistency over availability, but this document assumes that availability
-is preferred to total consistency.
+# Consistency Mechanisms
 
-# Consistency and Correctness at Key Acquisition
+There are a variety of ways in which reliant systems may build key consistency solutions,
+ranging in operational complexity to ease-of-implementation. In this section, we survey
+a number of possible solutions. The viability of each varies depending on the applicable
+threat model, external dependencies, and overall reliant system's requirements.
 
-There are a variety of ways in which reliant systems may build key consistency and correct systems (KCCS),
-ranging in operational complexity to ease-of-implementation. In this section, we survey a number of
-possible solutions. The viability of each varies depending on the applicable threat model, external
-dependencies, and overall reliant system's requirements.
+In each mechanism, the client has as input a candidate key and seeks to determine
+if it has a (globally) consistent version of the key.
```

Comment:
Is this the function we want to provide, or is it a part of the key acquisition process instead "get_key(..., consistent=true)" ?

I tend to think that an adjunct process is a building block of the acquisition process.  That is, you can take "get_key(..., consistent=false)" and add "is_consistent(k)" to upgrade the acquisition process, but that is not the only way to achieve a consistent outcome.

As the emphasis below is on acquisition, it would seem like the "get_key" approach is a better fit for this document, at least.

Reply:
I view this document as providing guidance for implementing "is_consistent(k)", since getting a key tends to be a protocol and deployment specific process. The emphasis on acquisition mainly exists because one would presumably want to run this check immediately after acquiring a key, not _during_ the process of getting the key. I'll make some changes to reflect this.